types/src/event.rs

```diff
@@ -229,6 +233,64 @@ impl EventHandle {
         }
     }
 }
+
+#[derive(::serde::Deserialize)]
+#[serde(rename = "EventKey")]
+struct EventKeyValue(
```

Comment:
.. how about doing custom Deserialize / Serialize impls, where you read / write u8 40 times?

Or update BCS to support up to, e.g., 64 items in an array?